core/src/main/java/hudson/PluginManager.java

```diff
@@ -327,6 +322,35 @@ protected void reactOnCycle(PluginWrapper q, List<PluginWrapper> cycle)
                                 }
                             });
 
+
+                            g.followedBy().attains(PLUGINS_LISTED).add("Checking plugin dependencies", new Executable() {
+                                @Override
+                                public void run(Reactor reactor) throws Exception {
+                                    for (PluginWrapper plugin: getPlugins()) {
+                                        String core = plugin.getManifest().getMainAttributes().getValue("Jenkins-Version");
+                                        if (core != null && Jenkins.getVersion().compareTo(new VersionNumber(core)) < 0) {
+                                            String s = "Plugin " + plugin.getShortName() + " required jenkins " + core + " or later.";
+                                            LOGGER.severe(s);
+                                            NOTICE.add(s);
+                                        }
+
+                                        for (Dependency dep : plugin.getDependencies()) {
+                                            if (dep.optional) continue;
+                                            PluginWrapper p = getPlugin(dep.shortName);
+                                            if (p == null) {
+                                                String s = "Plugin " + plugin.getShortName() + " is missing required dependency " + p.getShortName();
```

Comment:
Users are generally not shown plugin short names and shouldn't here either (at least not *only* short names).